tools/arm_pack_manager/__init__.py

```diff
@@ -1,309 +1,104 @@
-        stdout.flush()
-
-    def cache_file (self, url) :
-        """Low level interface to caching a single file.
-
-        :param url: The URL to cache.
-        :type url: str
-        :rtype: None
-        """
-        if not self.silent : print("Caching {}...".format(url))
-        dest = join(self.data_path, strip_protocol(url))
-        try :
-            makedirs(dirname(dest))
-        except OSError as exc :
-            if exc.errno == EEXIST : pass
-            else : raise
+    def __init__(self, silent, no_timeouts):
+        self._cache = _Cache(
+            silent, no_timeouts,
+            json_path=LocalPackDir, data_path=LocalPackDir
+        )
         try:
-            with open(dest, "wb+") as fd :
-                fd.write(urlopen(url).read())
-        except URLError as e:
-            stderr.write(e.reason)
-        self.counter += 1
-        self.display_counter("Caching Files")
-
-    def pdsc_to_pack (self, url) :
-        """Find the URL of the specified pack file described by a PDSC.
-
-        The PDSC is assumed to be cached and is looked up in the cache by its URL.
-
-        :param url: The url used to look up the PDSC.
-        :type url: str
-        :return: The url of the PACK file.
-        :rtype: str
-        """
-        content = self.pdsc_from_cache(url)
-        new_url = content.package.url.get_text()
-        if not new_url.endswith("/") :
-            new_url = new_url + "/"
-        return (new_url + content.package.vendor.get_text() + "." +
-                content.package.find('name').get_text() + "." +
-                largest_version(content) + ".pack")
-
-    def cache_pdsc_and_pack (self, url) :
-        self.cache_file(url)
-        try :
-            self.cache_file(self.pdsc_to_pack(url))
-        except AttributeError :
-            stderr.write("[ ERROR ] {} does not appear to be a conforming .pdsc file\n".format(url))
-            self.counter += 1
-
-    def get_urls(self):
-        """Extract the URLs of all know PDSC files.
-
-        Will pull the index from the internet if it is not cached.
-
-        :return: A list of all PDSC URLs
-        :rtype: [str]
-        """
-        if not self.urls :
-            try : root_data = self.pdsc_from_cache(RootPackURL)
-            except IOError : root_data = self.cache_and_parse(RootPackURL)
-            self.urls = ["/".join([pdsc.get('url').strip("/"),
-                                   pdsc.get('name').strip("/")])
-                         for pdsc in root_data.find_all("pdsc")]
-        return self.urls
+            self._legacy_names = load(open(LocalPackLegacyNames))
+        except IOError:
+            self._legacy_names = {}
 
     def _get_sectors(self, device):
         """Extract sector sizes from device FLM algorithm
 
-        Will return None if there is no algorithm, pdsc URL formatted in correctly
+        Will return None if there is no algorithm, pdsc URL formatted in
```

Comment:
Not sure why, but this feels like weird english.

Is it missing an `or`?

Reply:
Yes.